### DISCUSSION TITLE
Allow setting the cookie domain with SessionMiddleware for sharing cookies between subdomains

### DISCUSSION THREAD

**Discussion by johtso** [encode/starlette#1603] [Q&A] [answered]
Possibly this would just be a boolean option that if set gets the domain from the current request and uses that in the cookie?

**Comment by datasushi** [comment]
Yes please, looking for the same. But not boolean, make it string type so I can set it to what I need.

**Comment by jvelo** [comment]
Is there interest in supporting such a feature ? I have a fork handy that I can turn in a PR if there is.

IMO it makes sense to support it since host-based routing is supported.

**Comment by zllvm** [comment]
I've created a [PR2280](https://github.com/encode/starlette/pull/2280)

**Reply by alex-oleshkevich** [reply, marked as answer]
You can also use this lib: https://github.com/alex-oleshkevich/starsessions
